notebook/use_guide.clj

```diff
@@ -91,7 +91,9 @@ Playwright: This is a synopsis for the above play:
 (def synopsis
   (gen/complete-template
     synopsis-template
-    {:title "Mr. X" :genre "crime"}))
+    {:title "Mr. X" :genre "crime"}
+    {:api-key "sk-xxxxx"}
```

Comment:
a usage example of the complete-template function.
The user can the "code" whatever he want to get the key, bosquet does not need to prescribe this.

Reply:
One alternative would be to allow to "configure"  any "tag" in this way, so from a user perspective:

```
(gen/complete-template
    synopsis-template
    {:title "Mr. X" :genre "crime"}
    {:llm-generate
      {:api-key "sk-xxxxx"}}
```
this would allow to keep the same structure in case of more tags 
(but still assuming that all usages of the same tag in a multi-tag complex prompt get configured in the same way.)

Basically saying that all `llm-generate` tags in a complex prompt  **need** to use the same model. (at least when I want to externally configure it)
By "hardcoding" api-keys inside prompt definition this could still be done differently.

Reply:
A completely different way to achieve the same (external copnfiguraion of models), could be by allowing to use "global vars" inside a prompt definition, maye in this way:

```

(def synopsis-template
  "You are a playwright. Given the play's title and it's genre
it is your job to write synopsis for that play.

Title: {{title}}
Genre: {{genre}}

Playwright: This is a synopsis for the above play:
{% llm-generate model=text-davinci-003 
                          var-name=play
                          api-key=$$my-global-var-with-the-api-key$$ 

%}")

```

and a function to set the globals:
```
(set-global "my-global-var-with-the-api-key"   "sk-aaaaa")
```

Disadvantage here: The prompt definition would  decide which models opts to use, which might be different "per model". So the prompt definitions are less model agnostic.